drake/solvers/Optimization.h

```diff
@@ -0,0 +1,323 @@
+#ifndef DRAKE_OPTIMIZATION_H
+#define DRAKE_OPTIMIZATION_H
+
+#include "Core.h"
+#include <list>
+#include <memory>
+#include <initializer_list>
+
+namespace Drake {
+
+  struct DecisionVariable {
+    enum VarType {
+      CONTINUOUS,
+      INTEGER,
+      BINARY
+    };
+    VarType type;
+    std::string name;
+    Eigen::VectorXd value;
+    size_t start_index;
+  };
+  class DecisionVariableView {  // enables users to access pieces of the decision variables like they would any other eigen vector
+  public:
+    DecisionVariableView(const DecisionVariable& var) : var(var), start_index(0), length(var.value.rows()) {}
+    DecisionVariableView(const DecisionVariable& var, size_t start, size_t n) : var(var), start_index(start), length(n) {
+      assert(start+n<var.value.rows());
+    }
+
+    size_t index() const { return var.start_index + start_index; }
```

Comment:
Document this? It returns the index of the DicisionVariableView in the aggregate decision variables.